napari/utils/misc.py

```diff
@@ -260,3 +260,53 @@ def all_subclasses(cls: Type) -> set:
     return set(cls.__subclasses__()).union(
         [s for c in cls.__subclasses__() for s in all_subclasses(c)]
     )
+
+
+_have_warned_psutil = False
+
+
+def resize_dask_cache(nbytes: int = None):
+    """Create or resize the dask cache for opportunistic caching.
+
+    The cache object is an instance of a :class`cachey.Cache`, and is made
+    available at :attr:`napari.utils.dask_cache`.
+
+    See `Dask oportunistic caching
+    <https://docs.dask.org/en/latest/caching.html>`_
+
+    Parameters
+    ----------
+    nbytes : int, optional
+        The desired size of the cache, in bytes, by default, napari wil attempt
+        to import ``psutil`` and use 80% of the available memory.  Otherwise it
+        will emit a warning and use 32GB.
```

Comment:
This seems reasonable to me, but curious what others think

Reply:
Unless we want to avoid dependency creep, I'd also be quite happy to just depend on psutil.  as a "high performance" app, I can definitely imagine needing it in the future for system monitoring features.

Reply:
I'm happy for us to depend on psutil too - seems like a major and small package

Reply:
https://github.com/giampaolo/psutil

Reply:
I think 50% is a better target by default. I generally prefer having headroom with my RAM, rather than have a process grind my computer to a halt...

I'm ok with depending on psutil.

Reply:
sounds good.  how about i add another keyword argument that lets you state either the absolute amount, or the percentage (since we'll have psutil)?

Reply:
Sure!